idempotency_key/decorators.py

```diff
@@ -13,16 +13,35 @@
 #   ...
 
 
-def idempotency_key(view_func):
+def idempotency_key(*args, cache_name=None):
     """
-    Mark a view function as requiring idempotency key protection but the view should control the response.
+    Allows an optional cache name to be specified so that different cache settings can be used on a per-view function
+    basis.
+    :param args: optioanl arguments. This can contain the view function object if cache_name is not specified
+    :param cache_name: The name of the cache to use from the settings file under CACHES={...}
+    :return: wrapped function
     """
+    def _idempotency_key(view_func):
+        """
+        Mark a view function as requiring idempotency key protection but the view should control the response.
+        """
 
-    def wrapped_view(*args, **kwargs):
-        return view_func(*args, **kwargs)
+        @wraps(view_func)
+        def wrapped_view(*args, **kwargs):
+            return view_func(*args, **kwargs)
 
-    wrapped_view.idempotency_key = True
-    return wraps(view_func)(wrapped_view)
+        wrapped_view.idempotency_key = True
+        if cache_name:
+            wrapped_view.idempotency_key_cache_name = cache_name
```

Comment:
Maybe it's worth doing a check that the cache with the given name exists.
UPDATE: I saw that the library fails to process requests if the cache doesn't exist, maybe it'd be good to do the check to prevent a successful start and fail later when requests are sent.